### ISSUE TITLE
Add support for macOS to CompilerGym CMake build

### ISSUE THREAD

**Issue by ChrisCummins** [facebookresearch/CompilerGym#597]
## 🚀 Feature

CMake does not yet support building on macOS.

**Comment by chendian520** [comment]
I want to build gym from source ,but encoutered this problem :

CMake Error at build2/_deps/grpc-src/CMakeLists.txt:2381 (target_link_libraries):
Target "grpc" links to:

Copy
absl::any_invocable
but the target was not found. Possible reasons include:

Copy

There is a typo in the target name.
A find_package call is missing for an IMPORTED target.
An ALIAS target is missing.
Is there any solutions or suggetstions?

**Comment by ChrisCummins** [comment]
> I want to build gym from source ,but encoutered this problem :
> 
> CMake Error at build2/_deps/grpc-src/CMakeLists.txt:2381 (target_link_libraries): Target "grpc" links to:
> 
> Copy absl::any_invocable but the target was not found. Possible reasons include:

FWIW I'm not sure that that issue in particular is limited to macOS.

Cheers,
Chris

**Comment by chendian2023** [comment]
Yes, I encoutered this problem in linux ubuntu , do you have any suggetions? Thanks

**Comment by chendian2023** [comment]
what should I do if I had built grpc ,  use COMPILER_GYM_GRPC_PROVIDER= external   -DgRPC_DIR=/path/to/grpc/lib/cmake/grpc ??
like install.md :-DCOMPILER_GYM_LLVM_PROVIDER=external \
                    -DLLVM_DIR=/path/to/llvm/lib/cmake/llvm
but what about absl  ?How to set absl?  Because I  built gprc from source and also encoutered this problem
Found package configuration file:        
                                                                                                                      
    /home/chendian/tools/grpc-install/lib/cmake/grpc/gRPCConfig.cmake

  but it set gRPC_FOUND to FALSE so package "gRPC" is considered to be NOT
  FOUND.  Reason given by package:

  The following imported targets are referenced, but are missing:
  **absl::any_invocable**
 
thanks for your relpy!

**Comment by chendian2023** [comment]
cmake .. -DCOMPILER_GYM_GRPC_PROVIDER=external -DgRPC_DIR=/home/chendian/tools/grpc-install/lib/cmake/grpc -DCOMPILER_GYM_LLVM_PROVIDER=external -DLLVM_DIR=/home/chendian/tools/tmp/clang+llvm-10.0.0-x86_64-linux-gnu-ubuntu-18.04/lib/cmake/llvm -DCMAKE_INSTALL_PREFIX=/hom/chendian/tools/tmp/gym_install 2>&1 | tee llvm-grpcnew.log                 

what's more ,this is my build scripts

**Comment by chendian2023** [comment]
"-- Build files have been written to: /home/chendian/tools/tmp/CompilerGym/build-grpc-new2  "

After replace  absl and  grpc  . It seems that I had success in "cmake"  ,but there is no makefile in my build directory ,  what should I do next ?  There is no insructions in the install.md  after $ cmake ... \
    -DCOMPILER_GYM_LLVM_PROVIDER=external \
    -DLLVM_DIR=/path/to/llvm/lib/cmake/llvm  
  in 
 https://github.com/facebookresearch/CompilerGym/blob/development/INSTALL.md .

Should I ues ninja && ninja install ?
Thanks for your reply .

**Comment by ChrisCummins** [comment]
@chendian2023:

`cmake --build "<path to build directory>"`

as per the install instructions.
